### PR TITLE
Prevent PHP CS Fixer from fixing PHPStan cache files

### PR DESCRIPTION
**Pull request by julienfalque**

### CODE REVIEW

**Comment by SpacePossum** [comment]
Thank you @julienfalque.

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,10 @@
 /.phpunit.result.cache
 /box.json
 /composer.lock
+/dev-tools/composer.lock
+/dev-tools/bin/
+/dev-tools/phpstan/cache/
+/dev-tools/vendor/
```
Comment (keradus) [inline .gitignore:8]: @julienfalque why do we need it here as well? git should pick nested .gitignore automatically
```
ker@dus:~/github/FriendsOfPHP/PHP-CS-Fixer/dev-tools λ cat .gitignore 
/composer.lock
/bin/
/phpstan/cache/
/vendor/
```

File: .php-cs-fixer.dist.php [hunk 2]
```diff
@@ -23,6 +23,7 @@
 EOF;
 
 $finder = PhpCsFixer\Finder::create()
+    ->ignoreVCSIgnored(true)
```
Comment (keradus) [inline .php-cs-fixer.dist.php:26]: maybe we should request Symfony to support nested `.gitignore` files?

Comment (julienfalque) [reply .php-cs-fixer.dist.php:26]: I might try to submit the PR myself... If I find some spare time for that :)

Comment (julienfalque) [reply .php-cs-fixer.dist.php:26]: See https://github.com/symfony/symfony/pull/43150.

**Pull request merged**